### PR TITLE
MGMT-2191 update app-interface saas.yaml based on the assisted-instal…

### PR DESCRIPTION
**Pull request by oshercc**
…ler.yaml in assisted-installer-deployment

### CODE REVIEW

File: tools/update_saas_target.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    logging.info("cloneing repo:  {command}".format(command=command))
+    subprocess.check_output(command, shell=True)
+
+    with open(args.saas_file, "r") as f:
+        updated_source = yaml.load(f)
+
+    with open(SAAS_FILE_PATH, "r") as f:
+        updated_target = yaml.load(f)
+
+    updated_target['resourceTemplates'][0]['targets'] = updated_source['targets']
```
Comment (oshercc) [inline tools/update_saas_target.py:50]: remove all fields that are removed from the target

File: target_saas.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
+targets:
+  - namespace:
+      $ref: /services/assisted-installer/namespaces/assisted-installer-integration.yml
+    ref: master
+    upstream: openshift-assisted-service-gh-build-master
+    parameters:
```
Comment (gamli75) [inline target_saas.yaml:6]: please remove all parameters except:
      INSTALLER_IMAGE: registry-proxy.engineering.redhat.com/rh-osbs/openshift4-assisted-installer-rhel8:v4.6.0-19
      CONTROLLER_IMAGE: registry-proxy.engineering.redhat.com/rh-osbs/openshift4-assisted-installer-reporter-rhel8:v4.6.0-15
      AGENT_DOCKER_IMAGE: registry-proxy.engineering.redhat.com/rh-osbs/openshift4-assisted-installer-agent-rhel8:v4.6.0-15
      CONNECTIVITY_CHECK_IMAGE: registry-proxy.engineering.redhat.com/rh-osbs/openshift4-assisted-installer-agent-rhel8:v4.6.0-15
      INVENTORY_IMAGE: registry-proxy.engineering.redhat.com/rh-osbs/openshift4-assisted-installer-agent-rhel8:v4.6.0-15
      FREE_ADDRESSES_IMAGE: registry-proxy.engineering.redhat.com/rh-osbs/openshift4-assisted-installer-agent-rhel8:v4.6.0-15
      DHCP_LEASE_ALLOCATOR_IMAGE: registry-proxy.engineering.redhat.com/rh-osbs/openshift4-assisted-installer-agent-rhel8:v4.6.0-15
      OPENSHIFT_INSTALL_RELEASE_IMAGE: quay.io/ocpmetal/ocp-release:4.6.0-0.nightly-2020-08-31-220837
      IGNITION_GENERATE_IMAGE: quay.io/ocpmetal/assisted-ignition-generator:v1.0.9.4
      RELEASE_TAG: v1.0.9.4-ds

File: tools/update_saas_target.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+logging.getLogger().setLevel(logging.INFO)
+
+APP_INTERFACE_REPO = "https://gitlab.cee.redhat.com/{user}/app-interface.git"
+SAAS_FILE_PATH = "app-interface/data/services/assisted-installer/cicd/saas.yaml"
+COMMIT_MESSAGE = "Updating saas file"
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--saas_file", help="deployment yaml file to update", type=str,
+ default=os.path.join(os.path.dirname(__file__), "../target_saas.yaml"))
+parser.add_argument("--user", help="Git lab user (used for forked)")
```
Comment (gamli75) [inline tools/update_saas_target.py:25]: if we can get it from the user laptop - it will be great

File: tools/update_saas_target.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+parser.add_argument("--deployment_update", help="Commit and push deployment changes", action='store_true')
+parser.add_argument("--update", help="Update changes to app-interface", action='store_true')
```
Comment (gamli75) [inline tools/update_saas_target.py:27]: not clear enough what is the difference between --deplyment_update and --update

File: tools/update_saas_target.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+args = parser.parse_args()
```
Comment (gamli75) [inline tools/update_saas_target.py:28]: please add release_version parameter

File: tools/update_saas_target.py [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    with open(args.saas_file, "r") as f:
+        updated_source = yaml.load(f)
+
+    with open(SAAS_FILE_PATH, "r") as f:
+        updated_target = yaml.load(f)
+
+    updated_target['resourceTemplates'][0]['targets'] = updated_source['targets']
+
+    with open(SAAS_FILE_PATH, "w") as f:
+        yaml.dump(updated_target, f)
```
Comment (gamli75) [inline tools/update_saas_target.py:53]: what will happen if parameter X is in the source yaml and not exists in the target.yaml?
We should update all parameters if they exist and fail the script with the relevant error

File: tools/update_saas_target.py [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+    subprocess.check_output(command, shell=True, cwd="./app-interface")
+
+def deployment_update():
+    command = "git commit -am\'{}\'".format(COMMIT_MESSAGE)
+    logging.info("committing changes:  {command}".format(command=command))
+    subprocess.check_output(command, shell=True)
+
+    command = "git push origin HEAD:master"
+    logging.info("pushing changes:  {command}".format(command=command))
+    subprocess.check_output(command, shell=True)
```
Comment (gamli75) [inline tools/update_saas_target.py:72]: after running the git push - there is a message in the output with a link how to open a MR - please add it to the script output

File: tools/update_saas_target.py [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+# This will create a branch named <user>/saas_file_update on the forked app-interface repo
+# updated with the target described under saas file (default is target_saas.yaml)
+###########################################################################################
+
+logging.basicConfig(format='%(asctime)s %(message)s')
+logging.getLogger().setLevel(logging.INFO)
+
+APP_INTERFACE_REPO = "https://gitlab.cee.redhat.com/{user}/app-interface.git"
+SAAS_FILE_PATH = "app-interface/data/services/assisted-installer/cicd/saas.yaml"
+COMMIT_MESSAGE = "Updating saas file"
```
Comment (gamli75) [inline tools/update_saas_target.py:20]: if the user enters a release version - please add it to the commit message

**Comment by osherdp** [comment]
just regular maintenance

**Pull request closed without merging**